### DISCUSSION TITLE
how to bundle mitmproxy executable with its python script using pyinstaller

### DISCUSSION THREAD

**Discussion by Srijan67** [mitmproxy/mitmproxy#7252] [Q&A]
I am adding mitmproxy executable bundle in client side machine which runs on background but my issue is since I want to intercept and modify network traffic and the command for doing so require .py script `mitmproxy -s proxy.py` so how can I protect this file from user then because its executable require python file as well ? Example python script is:
from mitmproxy import http
from mitmproxy.tools.main import mitmdump

```
async def request(flow: http.HTTPFlow) -> None:
    if flow.request.method == 'GET':
            flow.response = http.Response.make(
            200,  
            b"Hello World",  
            {"Content-Type": "text/html","access-control-allow-origin": "*"},
        )

if __name__ == '__main__':
    mitmdump(['-p', '9999', '-s', 'proxy.py', '--set', 'client_replay_concurrency=-1', '--set', 'websocket=false'])
```
I bundled using pyinstaller command `pyinstaller --onefile proxy.py` but the above script finds for .py file and mitmproxy need to read it raw it understand it.

How can I protect this file from client machine. Is there a way to bundle it up on the build or something.

Please help me with this!

**Comment by mhils** [comment]
This is not really well supported. You can maybe fork mitmproxy and add it to the builtin addons in https://github.com/mitmproxy/mitmproxy/tree/main/mitmproxy/addons.

I would recommend to just not "protect" the file.

**Reply by Srijan67** [reply]
Got it! Thanks for the response 👍

**Reply by Srijan67** [reply]
@mhils Is there any chance that you can add some kind of internal encryption or a way in mitmproxy to handle the py script internally without exposing it as you can imagine how much it will help if someone wants to build its exe and then use it in a standalone environment.

**Reply by mhils** [reply]
Not going to happen, sorry.

**Reply by Srijan67** [reply]
Ok, No problem and thanks for ur efforts : )
